File: src/main/java/com/matag/admin/email/MatagEmailSender.java
package com.matag.admin.email;

import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Component;

import com.matag.admin.config.ConfigService;

import lombok.AllArgsConstructor;
import lombok.SneakyThrows;

@Component
@AllArgsConstructor
public class MatagEmailSender {
  private final JavaMailSender emailSender;
  private final ConfigService configService;

  @SneakyThrows
  public void send(String receiver, String subject, String body) {
    var mimeMessage = emailSender.createMimeMessage();
    var helper = new MimeMessageHelper(mimeMessage, "utf-8");
    helper.setTo(receiver);
    helper.setSubject(configService.getMatagName() + " - " + subject);
    helper.setText(body, true);
    emailSender.send(mimeMessage);
  }
}

File: src/main/kotlin/com/matag/admin/email/MatagEmailSender.kt
package com.matag.admin.email

import com.matag.admin.config.ConfigService
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.MimeMessageHelper
import org.springframework.stereotype.Component

@Component
class MatagEmailSender(
    private val emailSender: JavaMailSender,
    private val configService: ConfigService,
) {
    fun send(receiver: String, subject: String?, body: String) {
        val mimeMessage = emailSender.createMimeMessage()
        val helper = MimeMessageHelper(mimeMessage, "utf-8")
        helper.setTo(receiver)
        helper.setSubject(configService.matagName + " - " + subject)
        helper.setText(body, true)
        emailSender.send(mimeMessage)
    }
}
